walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,78 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
 
+/** Finds options for times to schedule a meeting request */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // no options for too long of a request
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration())
+    {
+        Collection<TimeRange> noTimes = Arrays.asList();
+        return noTimes;
+    }
+
+    Collection<String> attendees = new HashSet<>();
+    attendees = request.getAttendees();
+
+    // initialize options as the whole day
+    Collection<TimeRange> times = new ArrayList<>();
+    times.add(TimeRange.WHOLE_DAY);
+
+    // if there are no attendees the meeting can be any time of the day 
+    if(attendees.isEmpty()){
+        return times; 
+    }
+    
+    for(Event event : events){
+
+        Collection<String> eventAttendees = new HashSet<>();
+        eventAttendees = event.getAttendees();
+        boolean relevantMeeting = false;
+
+        // a relevant event contains a person from the requested meeting
+        for(String person : eventAttendees){
+            if(attendees.contains(person)){
+                relevantMeeting = true;
+            }
+        }
+
+        if(relevantMeeting){
+
+            // find options that overlap with the relevant meeting
+            Collection<TimeRange> overlapTimes = new ArrayList<>();
+            for(TimeRange time : times){
+                if(time.overlaps(event.getWhen())){
+                    overlapTimes.add(time);
+                }
+            }
+
+            for(TimeRange overlapTime : overlapTimes){
+
+                // add an option for free time before the relevant meeting
```

Comment:
Is this a TODO comment? If so please begin it with "TODO: ", so it doesn't look like documentation of what the code currently does. (I don't see anything like such an "option" in the following code.)

Reply:
This is not a TODO. By option I meant a time that the meeting could occur (so one of the times that is returned in the collection). Should I clear up that comment?

Reply:
Please do reword it. I read "an option for free time before the meeting" and thought you meant "the option to ensure a break (e.g. 15 minutes) between meetings"

Reply:
Okay. I reworded to "add a meeting time to 'times' list that occurs before the conflict"